src/sentry/workflow_engine/service/action/impl.py

```diff
@@ -26,3 +27,29 @@ def update_action_status_for_organization_integration(
             integration_id=integration_id,
             dataconditiongroupaction__condition_group__organization_id=organization_id,
         ).update(status=status)
+
+    def update_action_status_for_sentry_app_via_uuid(
+        self,
+        *,
+        organization_id: int,
+        status: int,
+        sentry_app_install_uuid: str,
+    ) -> None:
+        Action.objects.filter(
+            type=Action.Type.SENTRY_APP,
+            config__sentry_app_identifier=SentryAppIdentifier.SENTRY_APP_INSTALLATION_UUID,
+            config__target_identifier=sentry_app_install_uuid,
```

Comment:
Do we have an index on these fields? If not, should we? This might be a costly query otherwise.

Reply:
@wedamija https://redash.getsentry.net/queries/9635/source < 1000 at most. It seems like the query isn't that expensive at the moment too

Reply:
Ok, usually we avoid doing these kinds of updates in bulk, but since it's a small number of rows it might be ok. Is there a chance that this could grow to 10ks-100ks of thousands in the future though? If so, probably we should process this in batches instead

Reply:
Errr it's possible in the way future, I can put a backlog ticket for that